src/lib/ciul/checksum.c

```diff
@@ -295,6 +295,61 @@ uint32_t ef_tcp_checksum_ipx(int af, const void* ipx, const struct tcphdr* tcp,
     return ef_tcp_checksum((const struct iphdr*)ipx, tcp, iov, iovlen);
 }
 
+int ef_udp_checksum_is_correct(const struct iphdr* ip, const struct udphdr* udp,
+                               const struct iovec* iov, int iovlen)
+{
+  uint64_t csum64;
+  uint32_t csum;
+
+  csum64 = (uint64_t)ip->saddr + ip->daddr +    /* This is the UDP */
```

Comment:
UDPv4 packets are allowed to omit their checksum. You either need to handle that case here, or call out in the documentation that the caller needs to handle it.